secrets/provider/kubernetes/secrets.go

```diff
@@ -156,5 +245,5 @@ func (k k8sStore) DeleteContent(ctx context.Context, providerId string) error {
 
 // SaveContent implements SecretsStore.
 func (k k8sStore) SaveContent(ctx context.Context, uri *secrets.URI, revision int, value secrets.SecretValue) (string, error) {
-	return k.broker.SaveJujuSecret(ctx, uri, revision, value)
+	return k.broker.SaveJujuSecret(ctx, uri.Name(revision), value)
```

Comment:
uri.Name() is only used once I think. And elsewhere we still use fmt.Printf("%s-%s"), eg in metaslice Names()
So perhaps the extra method doesn't really pay off and we can do without it.

Reply:
Both `k8sStore.SaveContent` and `vaultStore.SaveContent` uses this. And future providers should use this as well.

Reply:
Now we always use `uri.Name` to get the secret resource name.